docs/generators/go.md

```diff
@@ -8,6 +8,7 @@ sidebar_label: go
 |enumClassPrefix|Prefix enum with class name| |false|
 |hideGenerationTimestamp|Hides the generation timestamp when files are generated.| |true|
 |isGoSubmodule|whether the generated Go module is a submodule| |false|
+|legacyGoEnumNaming|whether to use legacy enum prefixes and suffixes which are shorter but subject to namespace collisions| |false|
```

Comment:
@fantavlik thanks for the PR. How's this option different from the`enumClassPrefix` option?

Reply:
Ah, good catch I think `enumClassPrefix` must have come along while this PR was in flight. I will remove `legacyGoEnumNaming` and rework my logic to obey the existing `enumClassPrefix`.

Reply:
Done, I pulled the use of `enumClassPrefix` from templates and into `AbstractGoCodegen.java` here: https://github.com/OpenAPITools/openapi-generator/pull/3162/commits/f9d8093992049b7c021a8be62c588b388e795720